src/api/auth.ts

```
@@ -180,10 +180,25 @@ export const bearerAuthorizer: APIGatewayTokenAuthorizerHandler = async (event)
     throw new Error('Unauthorized'); // returns a 401
   }
   const sanitizedToken = authorizationToken.replace(/Bearer /gi, '');
-  const data = jwt.verify(
-    sanitizedToken,
-    process.env.AUTH_SECRET,
-  );
+  let data;
+  try {
+    data = jwt.verify(
+      sanitizedToken,
+      process.env.AUTH_SECRET,
+    );
+  } catch (e) {
+    // XXX: find a way to return specific error to frontend or make all errors Unauthorized?
```

I think it would be nice to inform the user the type of error, if possible

---

I need to test this, but if the error thrown is an instance of `Error`, the API Gateway will return the message inside to the user along with a 401 (i think). I don't like exposing internal errors to the user but we could do it.

---

I agree with not exposing internal errors, but we are in control of the error that will get thrown here, right?

We could just throw an error with the nessage we want inside the catch block